src/main/java/DirectoryEncryptor/SyncDirectoryProcessor.java

```diff
@@ -0,0 +1,95 @@
+package DirectoryEncryptor;
+
+import EncryptionAlgorithms.EncryptionAlgorithm;
+import EventsLogger.EncryptionLog4JLogger;
+import EventsLogger.EncryptionLogEventsArgs;
+import EventsLogger.EncryptionLogger;
+import EventsLogger.Events;
+import Exceptions.InvalidKeyException;
+import FileEncryptor.FileEncryptor;
+import Keys.Key;
+import Utils.FileOperations;
+import org.apache.logging.log4j.Level;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Observable;
+
+public class SyncDirectoryProcessor extends DirectoryProcessor{
+    private final EncryptionLog4JLogger log4JLogger = new EncryptionLog4JLogger();
+
+    private String pathToDir;
+    private FileOperations directoryOperations;
+    private FileOperations encryptedDirectoryOperations;
+    private FileOperations decryptedDirectoryOperations;
+    private FileOperations keyFile;
```

Comment:
Remove FileOperations instances, use them in FileEncryptor instance for each file